ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/MinPriorityFeePerGasTransactionSelectorTest.java

```diff
@@ -0,0 +1,98 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.ethereum.blockcreation.txselection.BlockSelectionContext;
+import org.hyperledger.besu.ethereum.blockcreation.txselection.selectors.AbstractTransactionSelector;
+import org.hyperledger.besu.ethereum.blockcreation.txselection.selectors.MinPriorityFeePerGasTransactionSelector;
+import org.hyperledger.besu.ethereum.core.MiningParameters;
+import org.hyperledger.besu.ethereum.core.ProcessableBlockHeader;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.eth.transactions.PendingTransaction;
+import org.hyperledger.besu.plugin.data.TransactionSelectionResult;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class MinPriorityFeePerGasTransactionSelectorTest {
+  private AbstractTransactionSelector transactionSelector;
+
+  private final int minPriorityFeeParameter = 7;
+
+  @BeforeEach
+  public void initialize() {
+    MiningParameters miningParameters =
+        MiningParameters.newDefault().setMinPriorityFeePerGas(Wei.of(minPriorityFeeParameter));
+    BlockSelectionContext context =
+        new BlockSelectionContext(
+            miningParameters,
+            null,
+            null,
+            mock(ProcessableBlockHeader.class),
+            null,
+            null,
+            null,
+            null);
+    transactionSelector = new MinPriorityFeePerGasTransactionSelector(context);
+  }
+
+  @Test
+  public void shouldNotSelectWhen_PriorityFeePerGas_IsLessThan_MinPriorityFeePerGas() {
+    var transaction = mockTransactionWithPriorityFee(minPriorityFeeParameter - 1);
+    assertSelection(transaction, TransactionSelectionResult.PRIORITY_FEE_PER_GAS_BELOW_CURRENT_MIN);
+  }
+
+  @Test
+  public void shouldSelectWhen_PriorityFeePerGas_IsEqual_MinPriorityFeePerGas() {
+    var transaction = mockTransactionWithPriorityFee(minPriorityFeeParameter);
+    assertSelection(transaction, TransactionSelectionResult.SELECTED);
+  }
+
+  @Test
+  public void shouldSelectWhen_PriorityFeePerGas_IsGreaterThan_MinPriorityFeePerGas() {
+    var transaction = mockTransactionWithPriorityFee(minPriorityFeeParameter + 1);
+    assertSelection(transaction, TransactionSelectionResult.SELECTED);
+  }
+
+  @Test
+  public void shouldSelectWhenPrioritySender() {
+    var prioritySenderTransaction = mockTransactionWithPriorityFee(minPriorityFeeParameter - 1);
+    assertSelection(
+        prioritySenderTransaction,
+        TransactionSelectionResult.PRIORITY_FEE_PER_GAS_BELOW_CURRENT_MIN);
+    when(prioritySenderTransaction.hasPriority()).thenReturn(true);
+    assertSelection(prioritySenderTransaction, TransactionSelectionResult.SELECTED);
+  }
+
+  private void assertSelection(
```

Comment:
maybe `assertSelectionResult` is a better name?

Reply:
done